nixos/modules/system/boot/loader/grub/grub.nix

```diff
@@ -292,19 +293,6 @@ in
         '';
       };
 
-      extraInitrd = mkOption {
-        type = types.nullOr types.path;
-        default = null;
-        example = "/boot/extra_initramfs.gz";
-        description = ''
-          The path to a second initramfs to be supplied to the kernel.
-          This ramfs will not be copied to the store, so that it can
-          contain secrets such as LUKS keyfiles or ssh keys.
-          This implies that rolling back to a previous configuration
-          won't rollback the state of this file.
-        '';
-      };
-
```

Comment:
Looks good to me; I'll try to test soon. A nitpick: if we're going to remove the `extraInitrd` option then it'd be good to add a `mkRemovedOptionModule` so people get error messages that point them to the right place and we can avoid repeating #41608; e.g.:

```nix
{
  imports = [
    (mkRemovedOptionModule [ "boot" "loader" "grub" "extraInitrd" ] ''
      ...some brief pointer to the initrd secrets options/docs, maybe with conversion example...
    '')
  ];
}
```



Another option would be to just leave in support for `extraInitrd` for now to give people time to migrate; I don't have strong feelings about this since I don't use GRUB, and I guess breaking compatibility saves adding deprecation warnings. But the `zcat | cpio` dance to extract pre-baked secrets in this format is a little arcane, so it'd be nice to document it.

Reply:
Is this option fundamentally incompatible with the new mechanism ?
Also: this requires release notes, no just a warning, as it makes systems unbootable.

Reply:
It would be possible to keep the `extraInitrd` option, but I wanted to limit complexity and I don't know of any use case for it other than including secrets, which is far better served by the `boot.initrd.secrets` option.

I added the `mkRemovedOptionModule` warning, but I don't know how to explain the conversion process because I have never done it myself.

Reply:
Something like:

```
To migrate to the new initrd secrets system, extract the extraInitrd
archive into your main filesystem:

  # zcat /boot/extra_initramfs.gz | cpio -idvmD /etc/secrets/initrd
  /path/to/secret1
  /path/to/secret2
  # rm /boot/extra_initramfs.gz

then replace boot.loader.grub.extraInitrd with boot.initrd.secrets:

  boot.initrd.secrets = {
    "/path/to/secret1" = "/etc/secrets/initrd/path/to/secret1";
    "/path/to/secret2" = "/etc/secrets/initrd/path/to/secret2";
  };

See the boot.initrd.secrets option documentation for more information.
```